build.sbt

```diff
@@ -61,117 +62,109 @@ lazy val docs = (project in file("docs"))
-lazy val freestyleMonixJS  = freestyleMonix.js
+lazy val monixJVM = monix.jvm
+lazy val monixJS  = monix.js
 
-lazy val freestyleEffects = (crossProject in file("freestyle-effects"))
+lazy val effects = (crossProject in file("freestyle-effects"))
   .dependsOn(freestyle)
   .settings(name := "freestyle-effects")
   .jsSettings(sharedJsSettings: _*)
+  .crossDepSettings(commonDeps: _*)
 
-lazy val freestyleEffectsJVM = freestyleEffects.jvm
-lazy val freestyleEffectsJS  = freestyleEffects.js
+lazy val effectsJVM = effects.jvm
+lazy val effectsJS  = effects.js
 
-lazy val freestyleAsync = (crossProject in file("async/async"))
+lazy val async = (crossProject in file("async/async"))
   .dependsOn(freestyle)
   .settings(name := "freestyle-async")
   .jsSettings(sharedJsSettings: _*)
+  .crossDepSettings(commonDeps: _*)
 
-lazy val freestyleAsyncJVM = freestyleAsync.jvm
-lazy val freestyleAsyncJS  = freestyleAsync.js
+lazy val asyncJVM = async.jvm
+lazy val asyncJS  = async.js
 
-lazy val freestyleAsyncMonix = (crossProject in file("async/monix"))
-  .dependsOn(freestyle, freestyleAsync)
+lazy val asyncMonix = (crossProject in file("async/monix"))
+  .dependsOn(freestyle, async)
   .settings(name := "freestyle-async-monix")
-  .settings(
-    libraryDependencies ++= Seq(
-      %%("monix-eval"),
-      %%("monix-cats")
-    )
-  )
+  .crossDepSettings(
+    commonDeps ++ Seq(
+      %("monix-eval"),
+      %("monix-cats")
+    ): _*)
   .jsSettings(sharedJsSettings: _*)
 
-lazy val freestyleAsyncMonixJVM = freestyleAsyncMonix.jvm
-lazy val freestyleAsyncMonixJS  = freestyleAsyncMonix.js
+lazy val asyncMonixJVM = asyncMonix.jvm
+lazy val asyncMonixJS  = asyncMonix.js
 
-lazy val freestyleAsyncFs = (crossProject in file("async/fs2"))
-  .dependsOn(freestyle, freestyleAsync)
+lazy val asyncFs = (crossProject in file("async/fs2"))
+  .dependsOn(freestyle, async)
   .settings(name := "freestyle-async-fs2")
-  .settings(
-    libraryDependencies ++= Seq(
-      %%%("fs2-core"),
-      %%%("fs2-cats")
-    )
-  )
   .jsSettings(sharedJsSettings: _*)
+  .crossDepSettings(commonDeps ++ Seq(%("fs2-core"), %("fs2-cats")): _*)
 
-lazy val freestyleAsyncFsJVM = freestyleAsyncFs.jvm
-lazy val freestyleAsyncFsJS  = freestyleAsyncFs.js
+lazy val asyncFsJVM = asyncFs.jvm
+lazy val asyncFsJS  = asyncFs.js
 
-lazy val freestyleCache = (crossProject in file("freestyle-cache"))
+lazy val cache = (crossProject in file("freestyle-cache"))
   .dependsOn(freestyle)
-  .settings(
-    name := "freestyle-cache"
-  )
+  .settings(name := "freestyle-cache")
   .jsSettings(sharedJsSettings: _*)
+  .crossDepSettings(commonDeps: _*)
 
-lazy val freestyleCacheJVM = freestyleCache.jvm
-lazy val freestyleCacheJS  = freestyleCache.js
+lazy val cacheJVM = cache.jvm
+lazy val cacheJS  = cache.js
 
-lazy val freestyleCacheRedis = (crossProject in file("freestyle-cache-redis"))
-  .dependsOn(freestyle, freestyleCache)
+lazy val cacheRedis = (crossProject in file("freestyle-cache-redis"))
```

Comment:
_rediscala_ has no ScalaJS version, so I think a cross projects doesn't make much sense. That is why I changed it to a JVM only project in my branch.